src/measure_theory/function/strongly_measurable/ae_sequence.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2023 Alex Kontorovich and Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Heather Macbeth
+-/
+
+import measure_theory.function.strongly_measurable.basic
+
+/-!
+# Sequence of strongly measurable functions associated to a sequence of a.e.-strongly-measurable
+  functions
+
+We define here tools to prove statements about limits (infi, supr...) of sequences of
+`ae_strongly_measurable` functions.
+Given a sequence of a.e.-strongly-measurable functions `f : ι → α → β` with hypothesis
+`hf : ∀ i, ae_strongly_measurable (f i) μ`, and a pointwise property `p : α → (ι → β) → Prop` such
+that we have `hp : ∀ᵐ x ∂μ, p x (λ n, f n x)`, we define a sequence of strongly measurable functions
+`ae_strongly_seq hf p` and a measurable set `ae_strongly_seq_set hf p`, such that
+* `μ (ae_strongly_seq_set hf p)ᶜ = 0`
+* `x ∈ ae_strongly_seq_set hf p → ∀ i : ι, ae_strongly_seq hf hp i x = f i x`
+* `x ∈ ae_strongly_seq_set hf p → p x (λ n, f n x)`
+-/
+
+open measure_theory filter topological_space function set measure_theory.measure
+open_locale ennreal topological_space measure_theory nnreal big_operators
+
+open measure_theory
+open_locale classical
+
+variables {α β : Type*} {ι : Sort*}  [measurable_space α]
+  [topological_space β] {f : ι → α → β} {μ : measure α} {p : α → (ι → β) → Prop}
+
+/-- If we have the additional hypothesis `∀ᵐ x ∂μ, p x (λ n, f n x)`, this is a measurable set
+whose complement has measure 0 such that for all `x ∈ ae_strongly_seq_set`, `f i x` is equal to
+`(hf i).mk (f i) x` for all `i` and we have the pointwise property `p x (λ n, f n x)`. -/
+def ae_strongly_seq_set (hf : ∀ i, ae_strongly_measurable (f i) μ) (p : α → (ι → β) → Prop) :
+  set α :=
+(to_measurable μ {x | (∀ i, f i x = (hf i).mk (f i) x) ∧ p x (λ n, f n x)}ᶜ)ᶜ
+
+/-- A sequence of measurable functions that are equal to `f` and verify property `p` on the
+measurable set `ae_strongly_seq_set hf p`. -/
+noncomputable
+def ae_strongly_seq (hf : ∀ i, ae_strongly_measurable (f i) μ) (p : α → (ι → β) → Prop) :
+  ι → α → β :=
+λ i x, if (x ∈ ae_strongly_seq_set hf p) then ((hf i).mk (f i) x) else (⟨f i x⟩ : nonempty β).some
```

Comment:
Why not:
```suggestion
λ i x, if (x ∈ ae_strongly_seq_set hf p) then ((hf i).mk (f i) x) else f i x
```

Reply:
Or even
```suggestion
f
```
which I believe satisfies all the same proofs below.

Reply:
Yes it seems not to work; it breaks something later, for a reason I don't understand. It certainly seems that one should be able to give a junk value if `x` isn't in this set, and it escapes me why `(⟨f i x⟩ : nonempty β).some` works whereas `f i x` doesn't.

Reply:
By the way, we're also closely following the approach in `ae_measurable_sequence.ae_seq` and adapting to the `ae_strongly_seq` setting...

Reply:
> By the way, we're also closely following the approach in `ae_measurable_sequence.ae_seq` and adapting to the `ae_strongly_seq` setting...

Ah, I see what's happening now, thanks. Can you add a comment along these lines to an `## Implementation notes` heading in the module docstring?

Reply:
Done, thanks!